### PR TITLE
Inpainting API updates, test notebook, auto request retries

### PR DESCRIPTION
**Pull request by pharmapsychotic**

### CODE REVIEW

**Review by aletts — APPROVED** [review]

File: src/stability_sdk/animation.py [hunk 1]
```diff
@@ -59,7 +56,8 @@ class BasicSettings(param.Parameterized):
     width = param.Integer(default=512, doc="Output image dimensions. Will be resized to a multiple of 64.")
     height = param.Integer(default=512, doc="Output image dimensions. Will be resized to a multiple of 64.")
     sampler = param.ObjectSelector(default='K_euler_ancestral', objects=["DDIM", "PLMS", "K_euler", "K_euler_ancestral", "K_heun", "K_dpm_2", "K_dpm_2_ancestral", "K_lms", "K_dpmpp_2m", "K_dpmpp_2s_ancestral"])
-    model = param.ObjectSelector(default=DEFAULT_MODEL, objects=["stable-diffusion-v1-5", "stable-diffusion-512-v2-1", "stable-diffusion-768-v2-1", "stable-diffusion-depth-v2-0"])
+    model = param.ObjectSelector(default=DEFAULT_MODEL, objects=["stable-diffusion-v1-5", "stable-diffusion-512-v2-1", "stable-diffusion-768-v2-1", "stable-diffusion-depth-v2-0", "custom"])
+    custom_model = param.String(default="", doc="Identifier of custom model to use.")
```
Comment (kostarion) [inline src/stability_sdk/animation.py:60]: How does user know which custom models are available?

File: src/stability_sdk/client.py [hunk 2]
```diff
@@ -296,18 +298,25 @@ def generate(
...
         :param init_strength: Strength of the initial image
         :param init_noise_scale: Scale of the initial noise
         :param mask: Mask to use (0 for pixels to change, 255 for pixels to keep)
         :param masked_area_init: How to initialize the masked area
+        :param mask_fixup: Whether to restore the unmasked area after diffusion
         :param guidance_preset: Preset to use for CLIP guidance
         :param guidance_cuts: Number of cuts to use with CLIP guidance
         :param guidance_strength: Strength of CLIP guidance
-        :return: The generated image
+        :return: dict mapping artifact type to data
```
Comment (kostarion) [inline src/stability_sdk/client.py:313]: Shouldn't we preserve current `generate` behaviour for existing sdk users with additional `return_all=False` param like you did in your previous PR?

**Review by kostarion — APPROVED** [review]

File: src/stability_sdk/animation.py (hunk 1 above)
Comment (pharmapsychotic) [reply src/stability_sdk/animation.py:60]: For alpha test and partners we'll let them know model id's that go up for testing now.

File: src/stability_sdk/client.py (hunk 2 above)
Comment (pharmapsychotic) [reply src/stability_sdk/client.py:313]: The Api class is new to animation branch so there are no existing users of it yet besides animation notebook. The complicating factor for returning single image is it now supports # samples (with intention of deprecating StabilityInference and mapping its calls to this new Api)

**Pull request merged**